src/os/bluestore/BlueStore.cc

```diff
@@ -6546,7 +6551,15 @@ int BlueStore::open_db_environment(KeyValueDB **pdb, bool to_repair)
 
 int BlueStore::close_db_environment()
 {
-  _close_db_and_around();
+  // verify that the db is actually closed
+  if (db) {
+    RocksDBStore* rocks_db = dynamic_cast<RocksDBStore*>(db);
+    ceph_assert(rocks_db);
+    ceph_assert(!rocks_db->is_open());
+    delete(db);
```

Comment:
IMO 
delete db;
is a better (more common) format..

Reply:
> I think it makes sense to add this new commit to my original PR once we have a final version... @aclamk - are you fine with that?
Please add it. Without your code this change does not make much sense.

Reply:
done, this one can be closed